c10/core/impl/alloc_cpu.cpp

```diff
@@ -70,6 +70,31 @@ void* alloc_cpu(size_t nbytes) {
       "DefaultCPUAllocator: not enough memory: you tried to allocate ",
       nbytes,
       " bytes.");
+#elif defined(__linux__)
```

Comment:
Hmm, should this change be limited to only x86 Linuxes? (I.e. I'm not sure if HugeTLB is enabled on ARM by default) 
Also, wouldn't one need to check that they are enabled by checking value of `/sys/kernel/mm/transparent_hugepage/enabled`?
```suggestion
#elif defined(__linux__) && defined(__x86_64__)
```

Reply:
Hi @malfet , In fact I've primarily tested this feature on aarch64, the THP and HugeTLBs are fully supported on aarch64. 
And regarding checking the sys config, as you know the valid values for 
`/sys/kernel/mm/transparent_hugepage/enabled` are `always`, `madvise`, `never`

For most of the distros the default value is `madvise` for which we need to explicitly set the madvise(,,MADV_HUGEPAGE). that's what the patch does.
I have also tested this feature with 'always' and 'never' settings and the feature works as expected: 
for 'always' THP would be always enabled, and the explicit madvise() is redundant, but doesn't throw any error
for 'never' this memory advise hint is ignored.